pkg/rancher-desktop/backend/backendHelper.ts

```diff
@@ -217,4 +227,124 @@ export default class BackendHelper {
+    let manifest = '';
+
+    for (const shim in await BackendHelper.containerdShims(vmx)) {
+      manifest += `---
+apiVersion: node.k8s.io/v1
+kind: RuntimeClass
+metadata:
+  name: ${ shim }
+handler: ${ shim }
+`;
+    }
+
+    if (manifest === '') {
+      // We delete the manifest file, but we don't actually delete old runtime classes in k3s that no longer exist.
+      // They won't work though, as the symlinks in /usr/local/bin have been removed.
+      await vmx.execCommand({ root: true }, 'rm', '-f', MANIFESTS_RUNTIMES_YAML);
+    } else {
+      await vmx.execCommand({ root: true }, 'mkdir', '-p', path.dirname(MANIFESTS_RUNTIMES_YAML));
+      // Newer k3s versions define all possible known runtimes; they don't really hurt,
+      // but may be confusing if the corresponding shim is not installed.
+      // Don't use path.join() here; it will use backslashes on Windows.
+      await vmx.execCommand({ root: true }, 'touch', `${ path.dirname(MANIFESTS_RUNTIMES_YAML) }/runtimes.yaml.skip`);
+      await vmx.writeFile(MANIFESTS_RUNTIMES_YAML, manifest, 0o644);
+    }
+  }
+
+  /**
+   * Install containerd-wasm shims into /usr/local/containerd-shims (and symlinks into /usr/local/bin).
+   */
+  static async installContainerdShims(vmx : VMExecutor, configureWASM : boolean) {
+    // Calling install-containerd-shims without source dirs will remove the symlinks from /usr/local/bin.
+    const sourceDirs : string[] = [];
+
+    if (configureWASM) {
+      sourceDirs.push(
+        // Copy shims bundled with the app itself first, user-managed shims may override.
+        path.join(paths.resources, 'linux', 'internal'),
+        paths.containerdShims,
+      );
+    }
+    await vmx.execCommand({ root: true }, 'mkdir', '-p', '/root');
+    await vmx.writeFile('/root/install-containerd-shims', INSTALL_CONTAINERD_SHIMS_SCRIPT, 'a+x');
+    await vmx.execCommand({ root: true }, '/root/install-containerd-shims', ...sourceDirs);
+  }
+
+  /**
+   * Write the containerd config file. If WASM is enabled, include a runtime definition
+   * for each installed containerd shim.
+   */
+  static async writeContainerdConfig(vmx : VMExecutor, configureWASM : boolean) : Promise<void> {
+    let config = CONTAINERD_CONFIG;
+
+    if (configureWASM) {
+      const shims = await BackendHelper.containerdShims(vmx);
+
+      for (const shim in shims) {
+        config += '\n';
+        config += `[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.${ shim }]\n`;
+        config += `  runtime_type = "/usr/local/bin/${ shims[shim] }"\n`;
+      }
+    }
+
+    await vmx.writeFile(CONTAINERD_CONFIG_TOML, config);
+  }
+
+  /**
+   * Configure the Moby containerd-snapshotter feature if WASM support is requested.
+   */
+  static async writeMobyConfig(vmx : VMExecutor, configureWASM : boolean) {
+    let config: Record<string, any>;
+
+    try {
+      config = JSON.parse(await vmx.readFile(DOCKER_DAEMON_JSON));
+    } catch (err: any) {
```

Comment:
It catches the case when the file does not exist (which right now is the default: the snapshotter setting is the only one we have so far). What other failure modes beyond the directory or the file not existing do you expect?

Reply:
`vmx.readFile` might fall over, because it seems like WSL would sometimes return empty strings for commands (which would be invalid JSON).

But yeah, I'm okay with it as-is I guess.